### PR TITLE
Update SubstrateCli to return String

### LINKED ISSUE #4663: CLI's version uses static str which prevents formatting
Original comment about the source of the issue: https://github.com/paritytech/cumulus/pull/42#discussion_r367934842

In this particular case we want to add text in the description that shows how to use the command line to pass parameters to the relay chain and the para chain. To do that it would be best if we know the name of the executable.

Unfortunately Version takes static str so it is not possible to format at runtime.

cc @bkchr

### PR DESCRIPTION
**Pull request by pscott**
Closes #4663 

`SubstrateCli` trait would return `&'static str` everywhere, making it hard to customize the messages. It now returns `String`, for easier messages!

### CODE REVIEW

File: bin/node-template/node/src/command.rs [hunk 1]
```diff
@@ -21,32 +21,32 @@ use crate::service;
 use sc_cli::{SubstrateCli, RuntimeVersion, Role, ChainSpec};
 
 impl SubstrateCli for Cli {
-	fn impl_name() -> &'static str {
-		"Substrate Node"
+	fn impl_name() -> String {
+		String::from("Substrate Node")
```
Comment (bkchr) [inline bin/node-template/node/src/command.rs:25]:
```suggestion
		"Substrate Node".into()
```

Here and everywhere else please

Comment (pscott) [reply bin/node-template/node/src/command.rs:25]: gonna change that on this pr and polkadot's and cumulus :)

Comment (pscott) [reply bin/node-template/node/src/command.rs:25]: wasn't sure what the best way to do that was

File: client/cli/src/lib.rs [hunk 2]
```diff
@@ -119,10 +127,13 @@ pub trait SubstrateCli: Sized {
...
+		let name = Self::executable_name();
+		let author = Self::author();
+		let about = Self::description();
 		let app = app
-			.name(Self::executable_name())
-			.author(Self::author())
-			.about(Self::description())
+			.name(&*name)
+			.author(&*author)
+			.about(&*about)
```
Comment (cecton) [inline client/cli/src/lib.rs:136]:
```suggestion
			.name(name)
			.author(author.as_str())
			.about(about.as_str())
```
I usually prefer to use `.as_str()` as it is more descriptive of what it does. But it's just a taste, do what you prefer

File: client/cli/src/lib.rs [hunk 3]
```diff
@@ -178,10 +189,13 @@ pub trait SubstrateCli: Sized {
...
```
Comment (cecton) [inline client/cli/src/lib.rs:198]:
```suggestion
			.name(name)
			.author(author.as_str())
			.about(about.as_str())
```

Comment (pscott) [reply client/cli/src/lib.rs]: https://github.com/paritytech/substrate/pull/6550/commits/66a8c331a69f4e2f226ce0328aabe73cabc57451

File: client/cli/src/lib.rs (hunk 2 above)
Comment (pscott) [reply client/cli/src/lib.rs]: https://github.com/paritytech/substrate/pull/6550/commits/66a8c331a69f4e2f226ce0328aabe73cabc57451

**Comment by pscott** [comment]
I have no idea why this job fails. My [Polkadot companion PR](https://github.com/paritytech/polkadot/pull/1333) is up to date with master, and I believe this pull is up to date with Substrate master. Any ideas? :)

**Comment by cecton** [comment]
Seems to be the same issue than #6543

**Comment by andresilva** [comment]
Having the same problem with the polkadot companion over at #6362.

**Comment by cecton** [comment]
Ah there is a `cargo update` right before the build. I suspect a dependency upgrade https://github.com/paritytech/substrate/blob/master/.maintain/gitlab/check_polkadot_companion_build.sh#L115

**Comment by TriplEight** [comment]
@cecton did it help?

**Comment by cecton** [comment]
That's not my PR actually :grin: @pscott I will make a small change to your branch for testing something with the CI

**Pull request merged**